staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/generated.proto

```diff
@@ -28,6 +28,15 @@ import "k8s.io/apimachinery/pkg/runtime/schema/generated.proto";
 // Package-wide variables from generator "generated".
 option go_package = "v1";
 
+// CELValidationRules is used for expression validation.
+// Rule represents the validation rule which will be evaluated by CEL.
+// Message represents the message displayed when validation failed.
+message CELValidationRules {
+  optional string rule = 1;
```

Comment:
Rule should be required?

Reply:
`required` seems not suggested in proto and proto3 dropped required(ref: https://github.com/protocolbuffers/protobuf/issues/2497#issuecomment-267422550)

Reply:
Gotcha. Should we instead of a validation rule to require the field? I don't think we want CRD authors creating validation rules that don't have a rule.

Reply:
I think this [line](https://github.com/kubernetes/kubernetes/blob/c9b7552ab17d594d92ece089d80f5738672e5280/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/validation/validation.go#L258) make sure the empty validator will not get into postProcess.

Reply:
Can we add a test showing that an attempt to add an empty validator list entry has the right behavior?